src/Gui/View3DInventorViewer.cpp

```diff
@@ -3117,18 +3082,49 @@ SbBool View3DInventorViewer::isAnimating() const
     return navigation->isAnimating();
 }
 
-/*!
- * Starts programmatically the viewer in animation mode. The given axis direction
- * is always in screen coordinates, not in world coordinates.
+/**
+ * @brief Change the camera pose with an animation
+ *
+ * @param orientation The new orientation
+ * @param rotationCenter The rotation center
+ * @param translation An additional translation on top of the translation caused by the rotation around the rotation center
+ */
+void View3DInventorViewer::startAnimation(const SbRotation& orientation,
+                                          const SbVec3f& rotationCenter, const SbVec3f& translation, bool wait)
+{
+    // Currently starts a FixedTimeAnimation. If there is going to be an additional animation like
+    // FixedVelocityAnimation, check the animation type from a parameter and start the right animation
+
+    int duration = App::GetApplication()
+                       .GetParameterGroupByPath("User parameter:BaseApp/Preferences/View")
+                       ->GetInt("AnimationDuration", 250);
+
```

Comment:
IMO it's a bad idea to make this function dependent on a user parameter because it will be more complicated to run consecutive calls of viewPosition() with different speeds. Then instead of a boolean argument and int could be used to increase flexibility.

Reply:
> it will be more complicated to run consecutive calls of viewPosition() with different speeds.

I guess 61e419b88ffac8cdb9f874363270327a0ab3ed3c also fixes this.

Reply:
> I guess 61e419b also fixed this.

It's OK.